File: src/main/java/pro/sky/java/weatherhistoryapp/service/HistoryService.java
package pro.sky.java.weatherhistoryapp.service;

import pro.sky.java.weatherhistoryapp.domain.HistoryRecord;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface HistoryService {

    Mono<HistoryRecord> save(HistoryRecord record);

    Flux<HistoryRecord> find(int amount);

    Flux<HistoryRecord> findAll();

    Mono<Void> deleteAll();
}

File: src/main/kotlin/pro/sky/java/weatherhistoryapp/service/HistoryService.kt
package pro.sky.java.weatherhistoryapp.service

import pro.sky.java.weatherhistoryapp.domain.HistoryRecord

interface HistoryService {
    suspend fun save(record: HistoryRecord): HistoryRecord
    suspend fun find(amount: Int): List<HistoryRecord>
    suspend fun findAll(): List<HistoryRecord>
    suspend fun deleteAll()
}